src/npyffi/objects.rs

```diff
@@ -412,3 +558,53 @@ pub struct PyArray_DatetimeDTypeMetaData {
     pub base: NpyAuxData,
     pub meta: PyArray_DatetimeMetaData,
 }
+
+// npy_packed_static_string and npy_string_allocator are opaque pointers
+// consider extern types when they are stabilized
+// https://github.com/rust-lang/rust/issues/43467
+pub type npy_packed_static_string = c_void;
+pub type npy_string_allocator = c_void;
+pub type PyArray_DTypeMeta = PyTypeObject;
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct npy_static_string {
+    size: usize,
```

Comment:
Can you make the `size` and `buf` fields `pub` please? They are public in the numpy2 C-API, and we'll need access to them if we ever want to add support for variable-width string arrays

Reply:
Done in #442.